lib/mysql.js

```diff
@@ -374,929 +315,137 @@ MySQL.prototype._serializeObject = function(obj) {
- * @param {String[]} [models] A model name or an array of model names. If not present, apply to all models
- * @param {Function} [cb] The callback function
- */
-MySQL.prototype.autoupdate = function (models, cb) {
-  var self = this;
-
-  if ((!cb) && ('function' === typeof models)) {
-    cb = models;
-    models = undefined;
-  }
-  // First argument is a model name
-  if ('string' === typeof models) {
-    models = [models];
-  }
-
-  models = models || Object.keys(this._models);
-
-  async.each(models, function(model, done) {
-    if (!(model in self._models)) {
-      return process.nextTick(function() {
-        done(new Error('Model not found: ' + model));
-      });
-    }
-    var table = self.tableEscaped(model);
-    self.query('SHOW FIELDS FROM ' + table, function(err, fields) {
-      self.query('SHOW INDEXES FROM ' + table, function(err, indexes) {
-        if (!err && fields && fields.length) {
-          self.alterTable(model, fields, indexes, done);
-        } else {
-          self.createTable(model, done);
-        }
-      });
-    });
-  }, cb);
-
-};
-
-/*!
- * Create table
- * @param model
- * @param cb
+ * Get the place holder in SQL for values, such as :1 or ?
+ * @param {String} key Optional key, such as 1 or id
+ * @returns {String} The place holder
  */
-MySQL.prototype.createTable = function (model, cb) {
-  var metadata = this._models[model].settings[this.name];
-  var engine = metadata && metadata.engine;
-  var sql = 'CREATE TABLE ' + this.tableEscaped(model) +
-    ' (\n  ' + this.propertiesSQL(model) + '\n)';
-  if (engine) {
-    sql += 'ENGINE=' + engine + '\n';
-  }
-  this.query(sql, cb);
+MySQL.prototype.getPlaceholderForValue = function (key) {
+  return '?';
 };
 
-/**
- * Check if the models exist
- * @param {String[]} [models] A model name or an array of model names. If not present, apply to all models
- * @param {Function} [cb] The callback function
- */
-MySQL.prototype.isActual = function(models, cb) {
-  var self = this;
-  var ok = false;
-
-  if ((!cb) && ('function' === typeof models)) {
-    cb = models;
-    models = undefined;
-  }
-  // First argument is a model name
-  if ('string' === typeof models) {
-    models = [models];
-  }
-
-  models = models || Object.keys(this._models);
-  
-  async.each(models, function(model, done) {
-    var table = self.tableEscaped(model);
-    self.query('SHOW FIELDS FROM ' + table, function(err, fields) {
-      self.query('SHOW INDEXES FROM ' + table, function(err, indexes) {
-        self.alterTable(model, fields, indexes, function(err, needAlter) {
-          if (err) {
-            return done(err);
-          } else {
-            ok = ok || needAlter;
-            done(err);
-          }
-        }, true);
-      });
-    });
-  }, function(err) {
-    if (err) {
-      return err;
-    }
-    cb(null, !ok);
-  });
+MySQL.prototype.getCountForAffectedRows = function(model, info) {
+  var affectedRows = info && typeof info.affectedRows === 'number' ?
```

Comment:
Is this really mysql specific? Maybe have this be the default impl in the baseconnector